### ISSUE TITLE
ARIA Attribute Use

### ISSUE THREAD

**Issue by ed-richards** [jchristopher/searchwp-live-ajax-search#104]
The plugin adds various ARIA attributes to the search `input` element, some of which appear to be incorrect. 

I spotted this while performing a Google Lighthouse Audit. Under the 'Accessibility' tab it points out that there are mismatching ARIA attributes on the `input` element. This can be caused by the lack of a required ARIA role or using ARIA attributes on an element that doesn't support them. 

Running a demo through the 'axe' accessibility extension in Chrome suggests the following:
> ARIA attribute is not allowed: aria-expanded="false"

Running a demo through the 'ARIA Validator' Chrome extension also suggests there may be other ARIA attributes that are not supported by the `input` element, such as:
- aria-activedescendant
- aria-autocomplete

It would be great to see a fix which stops the injection of the ARIA attributes which aren't supported by `input` elements!

**Comment by ed-richards** [comment]
Some further useful information can be found here:

[https://dequeuniversity.com/rules/axe/3.2/aria-allowed-attr](https://dequeuniversity.com/rules/axe/3.2/aria-allowed-attr)

**Comment by ed-richards** [comment]
Are there any updates on the progress of this fix? Otherwise it would be useful to know if this is something planned for a later release? Thanks!

**Comment by jchristopher** [comment]
Hi @ed-richards thank you for bumping the thread. I've made some commits in https://github.com/jchristopher/searchwp-live-ajax-search/pull/107 with the following updates:

- `aria-expanded` handling has been moved to the results container element which better matches its purpose (as far as I can see).
- `aria-activedescendant` has been omitted primarily because it requires that in this case all results have an `id` but that would impose a new requirement on the results template. It has been omitted in favor of `aria-selected` handling when using keyboard arrows to navigate search results. I _think_ that still makes sense, but am open to hear additional feedback!
- I looked in to `aria-autocomplete` and it _looks_ like the intent is to add it to an `input` as per https://www.w3.org/TR/2017/REC-wai-aria-1.1-20171214/#aria-autocomplete so it has been left in place for the time being.

If there are additional optimizations to be made I'd love to hear them as I'd like this plugin to be as accessible as we can make it! Going forward we can open up individual Issues per a11y improvement to keep things a bit more segmented 👍 

Thank you for taking the time to provide feedback/suggestions!

**Comment by ed-richards** [comment]
@jchristopher Thanks so much for taking the time to look into those issues, I really appreciate it! The update has resolved all the accessibility notices which were being flagged in tools such as Google Lighthouse which is perfect 👍
